core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -386,6 +388,29 @@ public void nameResolverReturnsEmptyList() {
   }
 
   @Test
+  public void loadBalancerThrowsInHandleResolvedAddresses() {
+    RuntimeException ex = new RuntimeException("simulated");
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(false);
+    ManagedChannel channel = createChannel(nameResolverFactory, NO_INTERCEPTOR);
+    ClientCall<String, Integer> call = channel.newCall(method, CallOptions.DEFAULT);
+    call.start(mockCallListener, new Metadata());
+    assertEquals(1, loadBalancerFactory.balancers.size());
+    LoadBalancer<?> loadBalancer = loadBalancerFactory.balancers.get(0);
+    doThrow(ex).when(loadBalancer).handleResolvedAddresses(
+        Matchers.<List<ResolvedServerInfo>>anyObject(), any(Attributes.class));
+    nameResolverFactory.allResolved();
+    ArgumentCaptor<Status> statusCaptor = ArgumentCaptor.forClass(Status.class);
```

Comment:
There are a fair number of Captors here.  Maybe make a `@Captor` instance variable up top and reuse between tests?